pkg/controller/highperf/highperf_controller.go

```diff
@@ -0,0 +1,234 @@
+package highperf
+
+import (
+	"context"
+	"time"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+
+	networkclientset "k8s.io/cloud-provider-gcp/crd/client/network/clientset/versioned"
+	nodeutil "k8s.io/cloud-provider-gcp/pkg/util"
+	"k8s.io/cloud-provider-gcp/providers/gce"
+	controllersmetrics "k8s.io/component-base/metrics/prometheus/controllers"
+
+	networkv1 "k8s.io/cloud-provider-gcp/crd/apis/network/v1"
+	networkv1alpha1 "k8s.io/cloud-provider-gcp/crd/apis/network/v1alpha1"
+)
+
+type Controller struct {
+	gceCloud         *gce.Cloud
+	networkClientset networkclientset.Interface
+
+	nwInformer   cache.SharedIndexInformer
+	gnpInformer  cache.SharedIndexInformer
+	nodeInformer coreinformers.NodeInformer
+
+	queue workqueue.RateLimitingInterface
+}
+
+func NewHighPerfController(
+	gceCloud *gce.Cloud,
+	networkClientset networkclientset.Interface,
+	nwInformer cache.SharedIndexInformer,
+	gnpInformer cache.SharedIndexInformer,
+	nodeInformer coreinformers.NodeInformer) *Controller {
+
+	hc := &Controller{
+		gceCloud:         gceCloud,
+		networkClientset: networkClientset,
+		nwInformer:       nwInformer,
+		gnpInformer:      gnpInformer,
+		nodeInformer:     nodeInformer,
+		queue:            workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "highperf"),
+	}
+
+	// TODO (bmorettodama): Add event handlers for Network and GNP object DeleteFunc
+
+	nodeInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    nodeutil.CreateAddNodeHandler(hc.nodeAdd),
+		UpdateFunc: nodeutil.CreateUpdateNodeHandler(hc.nodeUpdate),
```

Comment:
you need to handle Deletes too, don't you?

Reply:
No, if the node is deleted we don't need to revert the high perf interface since they will no longer exist, so we don't care about them.
What we worry about is network or gnp deletion, so that we can revert the interfaces for nodes that are currently using that network before its deleted.